src/networking.c

```diff
@@ -1063,6 +1063,27 @@ void disconnectSlaves(void) {
     }
 }
 
+/* Check if there is any other slave waiting dumping RDB finished expect me.
+ * This function is useful to judge current dumping RDB can be used for full
+ * synchronization or not. */
+int anyOtherSlaveWaitRdb(client *except_me) {
+    listIter li;
+    listNode *ln;
+    int wait_rdb = 0;
+
+    listRewind(server.slaves, &li);
+    while((ln = listNext(&li))) {
+        client *slave = ln->value;
+        if (slave != except_me &&
+            slave->replstate == SLAVE_STATE_WAIT_BGSAVE_END)
+        {
+            wait_rdb = 1;
+            break;
```

Comment:
since there's no iterator to destroy, we can just return here and no need for the variable.